plugins/woocommerce/tests/e2e-pw/tests/smoke-tests/update-woocommerce.spec.js

```diff
@@ -68,68 +88,113 @@ test.describe.serial(
+			'Assert that "WooCommerce" is listed and active',
+			async () => {
+				await expect(
+					page.locator( '.plugin-title strong', {
+						hasText: /^WooCommerce$/,
+					} )
+				).toBeVisible();
+				await expect(
+					page.locator( '#deactivate-woocommerce' )
+				).toBeVisible();
+			}
+		);
+
+		if ( UPDATE_WC !== 'nightly' ) {
+			await test.step(
+				'Verify that WooCommerce was updated to the expected version',
+				async () => {
+					const api = new wcApi( {
+						url: baseURL,
+						consumerKey: process.env.CONSUMER_KEY,
+						consumerSecret: process.env.CONSUMER_SECRET,
+						version: 'wc/v3',
+					} );
+
+					const response = await api
+						.get( 'system_status' )
+						.catch( ( error ) => {
+							throw new Error( error.message );
+						} );
+
+					const wcPluginData = response.data.active_plugins.find(
+						( { plugin } ) =>
+							plugin === 'woocommerce/woocommerce.php'
+					);
+
+					expect( wcPluginData.version ).toEqual( UPDATE_WC );
+				}
 			);
+		}
+	} );
+
+	test( 'can run the database update', async ( { page } ) => {
+		const updateButton = page.locator( 'text=Update WooCommerce Database' );
+		const updateCompleteMessage = page.locator(
+			'text=WooCommerce database update complete.'
+		);
 
-			// Navigate to 'Installed Plugins' page
+		await test.step( "Navigate to 'Installed Plugins' page", async () => {
 			await page.goto( 'wp-admin/plugins.php', {
 				waitUntil: 'networkidle',
 			} );
+		} );
 
-			// Skip this test if the "Update WooCommerce Database" button didn't appear.
-			test.skip(
-				! ( await updateButton.isVisible() ),
-				'The "Update WooCommerce Database" button did not appear after updating WooCommerce. Verify with the team if the WooCommerce version being tested does not really trigger a database update.'
-			);
+		await test.step(
+			'Skip this test if the "Update WooCommerce Database" button did not appear',
+			async () => {
+				test.skip(
+					! ( await updateButton.isVisible() ),
+					'The "Update WooCommerce Database" button did not appear after updating WooCommerce. Verify with the team if the WooCommerce version being tested does not really trigger a database update.'
+				);
+			}
+		);
 
-			// If the notice appears, start DB update
+		await test.step( 'Notice appeared. Start DB update', async () => {
 			await updateButton.click();
 			await page.waitForLoadState( 'networkidle' );
+		} );
 
-			// Repeatedly reload the Plugins page up to 10 times until the message "WooCommerce database update complete." appears.
-			for (
-				let reloads = 0;
-				reloads < 10 && ! ( await updateCompleteMessage.isVisible() );
-				reloads++
-			) {
-				await page.goto( 'wp-admin/plugins.php', {
-					waitUntil: 'networkidle',
-				} );
-
-				// Wait 10s before the next reload.
-				await page.waitForTimeout( 10000 );
+		await test.step(
+			'Repeatedly reload the Plugins page up to 10 times until the message "WooCommerce database update complete." appears',
```

Comment:
Would the [Playwright polling](https://playwright.dev/docs/test-assertions#polling) feature be more suitable for this scenario rather than implementing a retry policy based on the amount of refreshes + timeout?

Reply:
Yeah makes sense to use `expect.poll` instead! Thanks a lot! ❤️
Fixed in 23ca08edf41094dd10e1e367190efbdcd1711eaa